btcclient/client_block_subscriber.go

```diff
@@ -56,7 +57,7 @@ func NewWithBlockSubscriber(cfg *config.BTCConfig, retrySleepTime, maxRetrySleep
 		if err != nil {
 			return nil, fmt.Errorf("failed to get BTC backend: %v", err)
 		}
-		if backend != rpcclient.BitcoindPost25 {
+		if !strings.EqualFold(backend.String(), rpcclient.BitcoindPost25.String()) {
 			return nil, fmt.Errorf("zmq is only supported by bitcoind, but got %v", backend)
 		}
```

Comment:
```suggestion
		if _, ok := backend.(rpcclient.BitcoindVersion); !ok {
			return nil, fmt.Errorf("zmq is only supported by bitcoind, but got %v", backend)
		}
```

Reply:
This check is to ensure we use bitcoind not btcd, since only bitcoind supports zmq that is necessary for the vigilante. So as long as the type matches it's okay. The version of bitcoind is not necessarily 25 or after

Reply:
This check is to ensure we use bitcoind not btcd, since only bitcoind supports zmq that is necessary for the vigilante. So as long as the type matches it's okay. The version of bitcoind is not necessarily 25 or after